src/sql/postgres.zig

```diff
@@ -0,0 +1,3960 @@
+            const length = try reader.length();
+            bun.assert(length >= 4);
+
+            this.* = .{
+                .name = try reader.readZ(),
+                .value = try reader.readZ(),
+            };
+        }
+
+        pub const decode = decoderWrap(ParameterStatus, decodeInternal).decode;
+    };
+
+    pub const BackendKeyData = struct {
+        process_id: u32 = 0,
+        secret_key: u32 = 0,
+        pub const decode = decoderWrap(BackendKeyData, decodeInternal).decode;
+
+        pub fn decodeInternal(this: *@This(), comptime Container: type, reader: NewReader(Container)) !void {
+            if (!try reader.expectInt(u32, 12)) {
+                return error.InvalidBackendKeyData;
+            }
+
+            this.* = .{
+                .process_id = @bitCast(try reader.int32()),
+                .secret_key = @bitCast(try reader.int32()),
+            };
+        }
+    };
+
+    pub const ErrorResponse = struct {
+        messages: std.ArrayListUnmanaged(FieldMessage) = .{},
+
+        pub fn format(formatter: ErrorResponse, comptime _: []const u8, _: std.fmt.FormatOptions, writer: anytype) !void {
+            for (formatter.messages.items) |message| {
+                try std.fmt.format(writer, "{}\n", .{message});
+            }
+        }
+
+        pub fn deinit(this: *ErrorResponse) void {
+            for (this.messages.items) |*message| {
+                message.deinit();
+            }
+            this.messages.deinit(bun.default_allocator);
+        }
+
+        pub fn decodeInternal(this: *@This(), comptime Container: type, reader: NewReader(Container)) !void {
+            var remaining_bytes = try reader.length();
+            if (remaining_bytes < 4) return error.InvalidMessageLength;
+            remaining_bytes -|= 4;
+
+            if (remaining_bytes > 0) {
+                this.* = .{
+                    .messages = try FieldMessage.decodeList(Container, reader),
+                };
+            }
+        }
+
+        pub const decode = decoderWrap(ErrorResponse, decodeInternal).decode;
+
+        pub fn toJS(this: ErrorResponse, globalObject: *JSC.JSGlobalObject) JSC.JSValue {
+            var b = bun.StringBuilder{};
+            defer b.deinit(bun.default_allocator);
+
+            for (this.messages.items) |msg| {
+                b.cap += switch (msg) {
+                    inline else => |m| m.utf8ByteLength(),
+                } + 1;
+            }
+            b.allocate(bun.default_allocator) catch {};
+
+            for (this.messages.items) |msg| {
+                var str = switch (msg) {
+                    inline else => |m| m.toUTF8(bun.default_allocator),
+                };
+                defer str.deinit();
+                _ = b.append(str.slice());
+                _ = b.append("\n");
+            }
+
+            return globalObject.createSyntaxErrorInstance("Postgres error occurred\n{s}", .{b.allocatedSlice()[0..b.len]});
```

Comment:
Its ok to just return the error instance instead of throwing?

Reply:
in this case, we throw a SyntaxError which becomes a promise rejection later